README.md

```diff
@@ -1,16 +1,166 @@
+    code=f"Id-{guid}",
+    base_currency="GBP",
+    created=datetime.datetime(2021, 3, 20, tzinfo=pytz.utc)
+)
+
+portfolio = tx_portfolios_api.create_portfolio(scope, create_transaction_portfolio_request=portfolio_request)
+portfolio_code = portfolio.id.code
+print("Porfolio Code:", portfolio_code)
+```
+
+### Upsert instruments
+
+```python
+instruments_api = api_factory.build(lusid.api.InstrumentsApi)
+
+# FIGI is from https://www.openfigi.com/id/BBG000C6K6G9
+figis_to_create = {
+    "BBG000C6K6G9": models.InstrumentDefinition(name="VODAFONE GROUP PLC",
+        identifiers={"Figi": models.InstrumentIdValue(value="BBG000C6K6G9")}
+    )
+}
+
+instruments_api.upsert_instruments(request_body=figis_to_create)
+```
+
+### Get instruments
+
+This example assumes that you have run the code samples under [upsert instruments](#upsert-instruments). You can retrieve the instruments and store them into two dictionaries (LUID->Name and Name->LUID) to use when interacting with other APIs:
+
+```python
+instruments_response = instruments_api.get_instruments(
+    identifier_type="Figi", request_body=list(figis_to_create.keys()))
+name_to_luid = {
+    value.name: value.lusid_instrument_id
+    for _, value in instruments_response.values.items()
+}
+luid_to_name = {v: k for k, v in name_to_luid.items()}
+```
+
+### Upsert transactions
+
+This example assumes that you have run the code samples under [create portfolio](#create-portfolio), [upsert instruments](#upsert-instruments), and [get instruments](#get-instruments).
+
+```python
+import uuid
+import datetime
+import pytz
+
+tx_portfolios_api = api_factory.build(lusid.api.TransactionPortfoliosApi)
+
+tx1 = models.TransactionRequest(
+    transaction_id=f"Transaction-{uuid.uuid4()}",
+    type="StockIn",
+    instrument_identifiers={"Instrument/default/LusidInstrumentId": name_to_luid["VODAFONE GROUP PLC"]},
+    transaction_date=datetime.datetime(2021, 3, 27, tzinfo=pytz.utc),
+    settlement_date=datetime.datetime(2021, 3, 28, tzinfo=pytz.utc),
+    units=100,
+    transaction_price=models.TransactionPrice(price=103),
+    total_consideration=models.CurrencyAndAmount(amount=103 * 100, currency="GBP"),
+    source="Broker"
+)
+
+tx_portfolios_api.upsert_transactions(scope=scope, code=portfolio_code, transaction_request=[tx1])
+```
+
+### Get holdings
+
+This example assumes that you have run the code samples under [create a portfolio](#create-portfolio), [upsert instruments](#upsert-instruments), [get= instruments](#get-instruments), and [upsert transactions](#upsert-transactions).
+
+```python
+import uuid
+import datetime
+import pytz
+
+tx_portfolios_api = api_factory.build(lusid.api.TransactionPortfoliosApi)
+
+holdings_response = tx_portfolios_api.get_holdings(
+    scope=scope, code=portfolio_code, property_keys=["Instrument/default/Name"]).values
+
+print("Holdings:")
+for holding in holdings_response:
+    print(luid_to_name[holding.instrument_uid], holding.units, holding.cost.amount)
+```
+
```

Comment:
is it worth creating another file that has all this code in that they can run? i notice you've got imports for each snippet but also refer to vars from previous snippets. Its not completely clear if we're suggesting these run individually or in one uber script. If its the latter, perhaps we should have something they can run?

Reply:
Yeh I think that makes sense. Ideally it could go in a script that we could include into the README, but GitHub doesn't support doing that. 

I initially wrote the code as one uber script but then thought it wasn't that easy to understand what was going on, so I pulled it apart. I wonder whether it would be simpler to say at the top that the snippets all rely on the preceding ones?

Reply:
Added file with all the code and linked to it.